pytest/tests/loadtest/locust/common/base.py

```diff
@@ -302,13 +302,17 @@ def poll_tx_result(self, meta: dict, tx: Transaction):
     def account_exists(self, account_id: str) -> bool:
         return "error" not in self.node.get_account(account_id, do_assert=False)
 
-    def create_contract_account(self,
-                                funding_account: Account,
-                                key: key.Key,
-                                balance: int = 50000):
-        self.send_tx_retry(
-            CreateSubAccount(funding_account, key, balance=balance),
-            "create contract account")
+    def prepare_account(self, account: Account, parent: Account, balance: int,
+                        msg: str) -> bool:
+        """
+        Create the account if it doesn't exist and refreshes the nonce.
+        """
+        exists = self.account_exists(account.key.account_id)
+        if not exists:
+            self.send_tx_retry(
+                CreateSubAccount(parent, account.key, balance=balance), msg)
+        account.refresh_nonce(self.node)
```

Comment:
Do we need to refresh the nonce for a new account?
If not, we can simplify the function to:
```python
if self.account_exists(account.key.account_id):
  account.refresh_nonce(self.node)
  return true

self.send_tx_retry(
    CreateSubAccount(parent, account.key, balance=balance), msg)
return false
```

Reply:
Yeah we actually have to refresh it. The Nonce is set to `(block_height - 1) * 1e6` at creation, see here: https://github.com/near/nearcore/blob/038eef905ed367aeb29d82ca1c2f6026c4754f28/runtime/runtime/src/actions.rs#L592-L593